waku/v2/waku_filter/rpc.nim

```diff
@@ -15,7 +15,7 @@ type
 
   FilterRequest* = object
     contentFilters*: seq[ContentFilter]
-    pubsubTopic*: string
+    pubsubTopic*: Option[string]
```

Comment:
Not sure I agree with where these changes are made. Let me explain:
This essentially changes the underlying protocol to have pubsubTopic/shard as optional, which would mean the _protocol_ would have to become aware of autosharding and mapping rather than this simply being a running use case for the application. In this case, for example, I think the filter service node would have to be able to receive a FilterRequest without `pubsubTopic` and somehow "know" about autosharding, even though it's most probably unaware of the application(s) it's servicing.
I think the "cleaner" (and simpler) way to introduce autosharding is to simply change the most external API (the one used by applications) to fill out shards/pubsubTopics where it's necessary but not populated by the application. This way the protocols remain untouched and still works as before.

Reply:
not sure if this is what you want or dont

> I think the filter service node would have to be able to receive a FilterRequest without pubsubTopic and somehow "know" about autosharding, even though it's most probably unaware of the application(s) it's servicing

imho pubsub topic should not be part of the request if content topic is already there, since it would be redundant. or am i missing something?